### PR TITLE
Adds checks and set to modsuit access and locking

### LINKED ISSUE #81997: MODsuit ID locks don't work at all
## Issue Summary

The MODsuit ID lock button doesn't do anything, It just toggles locked/unlocked regardless of ID

## Round ID:

N/A

## Testmerges:

N/A

## Reproduction:

Lock MODsuit with an ID, remove said ID, You can activate the MODsuit and unlock the ID lock anyways.
Also, Nukie MODsuits start locked and can't be activated without an agent ID but you can bypass that by just unlocking it.

### PR DESCRIPTION
**Pull request by Bilbo367**
## About The Pull Request

Lets you lock your modsuit. Also alerts you when you cant unlock it because you don't have the access.

## Why It's Good For The Game

Closes https://github.com/tgstation/tgstation/issues/81997

## Changelog

:cl:
fix: ID locking and unlocking work.
/:cl:

### CODE REVIEW

File: code/modules/mod/mod_ui.dm [hunk 1]
```diff
@@ -84,8 +84,13 @@
 		return
 	switch(action)
 		if("lock")
-			locked = !locked
-			balloon_alert(usr, "[locked ? "locked" : "unlocked"]!")
+			if(locked & allowed(usr) || !locked)
```
Comment (JohnFulpWillard) [inline code/modules/mod/mod_ui.dm:87]:
```suggestion
			if(!locked || allowed(usr))
```

File: code/modules/mod/mod_ui.dm [hunk 2]
```diff
@@ -84,8 +84,13 @@
...
+				req_access = wearer.get_access()
```
Comment (JohnFulpWillard) [inline code/modules/mod/mod_ui.dm:88]: This isn't doing anything

File: code/modules/mod/mod_ui.dm [hunk 3]
```diff
@@ -84,8 +84,13 @@
...
+				locked = !locked
+				balloon_alert(usr, "[locked ? "locked" : "unlocked"]!")
```
Comment (JohnFulpWillard) [inline code/modules/mod/mod_ui.dm:90]: Please replace ``usr`` in this proc with ``ui.user``.

File: code/modules/mod/mod_ui.dm (hunk 2 above)
Comment (Bilbo367) [reply code/modules/mod/mod_ui.dm:88]: 88 sets the req_access which is what's checked when attempting to activate the modsuit.

Comment (JohnFulpWillard) [reply code/modules/mod/mod_ui.dm:88]: We're not supposed to be updating the access here though, here is only flipping whether it's locked.

Comment (ZephyrTFA) [reply code/modules/mod/mod_ui.dm:88]: just dropping by to say that you are using the wrong comparison. That's the bitwise comparison not the logical comparison

Comment (JohnFulpWillard) [reply code/modules/mod/mod_ui.dm:88]: You're making it update the access upon locking/unlocking, that should not be the case. Access being set is in a separate place, the only thing we're supposed to do is change the status on whether it is locked, NOT set ``req_access`` to anything.

Comment (Bilbo367) [reply code/modules/mod/mod_ui.dm]: Ok i yield access will not be set when the modsuit user locks the modsuit.

**Pull request merged**